src/pre-engagement-form/index.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/* eslint-disable react/require-default-props */
+import React from 'react';
+import { connect } from 'react-redux';
+import { useForm, FormProvider } from 'react-hook-form';
+import * as FlexWebChat from '@twilio/flex-webchat-ui';
+
+import { AseloWebchatState } from '../aselo-webchat-state';
+import type { PreEngagementForm as PreEngagementFormDefinition } from './form-components/types';
+import { generateForm } from './form-components';
+import { LocalizationProvider } from './localization';
+import SubmitButton from './form-components/submit-button';
+import Title from './form-components/title';
+import { resetForm } from './state';
+
+export { PreEngagementFormDefinition };
+
+export const EMAIL_PATTERN = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
```

Comment:
Relocate this to be with the forms generators?

Reply:
I didn't really find a good place to put the `EMAIL_PATTERN` for now. I've left some `TODOs` to improve some areas related to validation and displaying error messages, I think it will be more clear where to place `EMAIL_PATTERN` when we do that.